openverse_catalog/dags/providers/provider_api_scripts/nappy.py

```diff
@@ -0,0 +1,123 @@
+
+from common import constants
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+logger = logging.getLogger(__name__)
+
+
+class NappyDataIngester(ProviderDataIngester):
+    providers = {constants.IMAGE: prov.NAPPY_DEFAULT_PROVIDER}
+    endpoint = "https://api.nappy.co/v1/openverse/images"
+    headers = {"User-Agent": prov.UA_STRING, "Accept": "application/json"}
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # On the first request, `prev_query_params` will be `None`. We can detect this
+        # and return our default params.
+        if not prev_query_params:
+            return {
+                "page": 1,
+            }
+        else:
+            return {
+                "page": prev_query_params["page"] + 1,
+            }
+
+    def get_batch_data(self, response_json):
+        # Takes the raw API response from calling `get` on the endpoint, and returns
+        # the list of records to process.
+        if response_json:
+            return response_json.get("images")
+        return None
+
+    def get_should_continue(self, response_json):
+        return bool(response_json.get("next_page"))
+
+    def get_media_type(self, record: dict):
+        return constants.IMAGE
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        # Parse out the necessary info from the record data into a dictionary.
+
+        if (foreign_landing_url := data.get("foreign_landing_url")) is None:
+            return None
+
+        if (image_url := data.get("url")) is None:
+            return None
+
+        # Hardoded to CC0, the only license Nappy.co uses
+        license_info = get_license_info(
+            "https://creativecommons.org/publicdomain/zero/1.0/"
+        )
+        if license_info is None:
+            return None
+
+        def _convert_filesize(raw_filesize_string: str) -> int:
+            """
+            Convert sizes from strings to byte integers, ex. "187.8kB" to 188.
+            """
+            filetype_multipliers = {"kB": 1000, "MB": 1_000_000, "GB": 1_000_000_000}
+            multiplier = filetype_multipliers[raw_filesize_string[-2:]]
+
+            return int(round(float(data.get("filesize")[:-2]) * multiplier))
+
+        # OPTIONAL FIELDS
+        # Obtain as many optional fields as possible.
+        foreign_identifier = data.get("foreign_identifier")
+        thumbnail_url = data.get("url") + "?auto=format&w=600&q=75"
+        filesize = _convert_filesize(data.get("filesize"))
+        filetype = data.get("filetype")
+        creator = data.get("creator")
+        creator_url = data.get("creator_url")
+        title = data.get("title")
+        meta_data = {
+            "views": data.get("views"),
+            "saves": data.get("saves"),
+            "downloads": data.get("downloads"),
+        }
```

Comment:
Thanks for adding these! We'll also want to add `downloads` [to the DDL](https://github.com/WordPress/openverse-catalog/blob/4d01826907dcf3a945e8d793783305a25cbd1ff9/docker/local_postgres/0004_openledger_image_view.sql#L11-L18) and the [image popularity metrics test](https://github.com/WordPress/openverse-catalog/blob/0728cdbe8b8ef4dc1037d877cfbf322f7252db1b/openverse_catalog/dags/common/popularity/sql.py#L44-L49)!